rust/blockstore/src/arrow/provider.rs

```diff
@@ -27,19 +27,19 @@ use uuid::Uuid;
 #[derive(Clone)]
 pub struct ArrowBlockfileProvider {
     block_manager: BlockManager,
-    sparse_index_manager: SparseIndexManager,
+    root_manager: RootManager,
 }
 
 impl ArrowBlockfileProvider {
     pub fn new(
         storage: Storage,
         max_block_size_bytes: usize,
         block_cache: Box<dyn PersistentCache<Uuid, Block>>,
-        sparse_index_cache: Box<dyn PersistentCache<Uuid, SparseIndex>>,
+        root_cache: Box<dyn PersistentCache<Uuid, RootReader>>,
```

Comment:
Leaving the comment here. Already discussed offline. It's ok if it is not addressable right now. It reads confusing that we are caching the reader instead of the data. Reader is a view of the data whereas data is data